dig.go

```diff
@@ -170,6 +176,78 @@ func (c *Container) provide(ctor interface{}, ctype reflect.Type) error {
 	return nil
 }
 
+func (c *Container) findConnections(n *node) (map[key]struct{}, error) {
+	var err error
+	keyPaths := make(map[key]string)
+	walkResult(n.Results, connectionVisitor{
+		c:    c,
+		n:    n,
+		err:  &err,
+		keys: keyPaths,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	keys := make(map[key]struct{}, len(keyPaths))
+	for k := range keyPaths {
+		keys[k] = struct{}{}
+	}
+	return keys, nil
+}
+
+type connectionVisitor struct {
+	c    *Container
+	n    *node
+	err  *error
+	keys map[key]string
+	path []string
```

Comment:
path can be ambiguous (I started thinking "import path").  Maybe `structLocation`, `currentAttributeIdentifier`? Something like that.

Reply:
Renamed to `currentResultPath` and documented that it's the path to the 'result' being visited.